src/screens/Bridge/Exchange-CrossChain/Exchange.tsx

```diff
@@ -226,13 +237,86 @@ function Exchange({ fetchExchangeTitle }: Props) {
 
   React.useEffect(() => {
     setSortOfferList([]);
+    setHideAllOffers(false);
+    setShowBestOffer(true);
   }, [fromValue, toAddress, fromAddress, chain]);
 
   const showOfferEstimateFailState = faileEstimateOffers === offers?.length;
   const ratesNotFound = toAsset && fromValue ? rate === 0 : false;
 
   const sortedOffers = isEmpty(sortOffersList) ? offers : sortingOffersToGasFee(sortOffersList);
 
+  // Use for select default best offer
+  React.useEffect(() => {
+    if (isEmpty(sortOffersList)) return;
+
+    const bestOffer = sortedOffers.find((offer) => !!offer.feeInfo && offer.provider !== EXCHANGE_PROVIDER.LIFI);
+    setSelectedProvider(bestOffer?.provider);
+
+    const failEstimatedFee = sortedOffers?.find((offer) => !offer.feeInfo);
+    if (!failEstimatedFee && showBestOffer) {
+      setHideAllOffers(true);
+    }
+  }, [sortedOffers, sortOffersList, renderItem]);
+
+  const exchangeOffers = React.useMemo(() => {
+    if (isEmpty(sortedOffers)) return [];
+
+    const bestOffer = sortedOffers.find((offer) => !!offer.feeInfo && offer.provider !== EXCHANGE_PROVIDER.LIFI);
+    return hideAllOffers ? [bestOffer] : sortedOffers;
+  }, [sortedOffers, offers, sortOffersList, renderItem]);
+
+  const selectedOffer = sortedOffers?.find((offer) => offer.provider === selectedProvider);
+  const validOffers = sortedOffers?.filter((offer) => !!offer.feeInfo);
+  const visibleOffers = !showLoading && !ratesNotFound && !isEmpty(sortedOffers) && !showOfferEstimateFailState;
+
+  const { errorMessage } = useTransactionFeeCheck(
+    chain || CHAIN.ETHEREUM,
+    selectedOffer?.feeInfo,
+    fromAsset,
+    selectedOffer?.fromAmount,
+  );
+
+  const confirmTransaction = () => {
+    if (!selectedOffer?.feeInfo) {
+      showTransactionRevertedToast();
+      return;
+    }
+
+    let transactionPayload: any = mapTransactionsToTransactionPayload(selectedOffer?.chain, selectedOffer.transactions);
```

Comment:
why `any` ? does `mapTransactionsToTransactionPayload` not return a response of a specific type?

Reply:
Updated it. `TransactionPayload` type

Reply:
Probably not as we're kinda fighting between two different typing supersets, typescript and flow - older code is written in flow, so the custom types won't be compatible with typescript